integration-tests/test-backcompat.sh

```diff
@@ -0,0 +1,34 @@
+#! /bin/sh
+
+set -e
+
+BACKWARD_IMAGE=v0.6.0
+
+. "$(dirname "$0")/common.sh"
+
+"$(dirname "$0")/start-components.sh"
+
+# Note we haven't created any tables in the DB yet - we aren't going to run long enough to flush anything
+
+echo Start first ingester, using older version
+docker run $RUN_ARGS -d --name=i1 --hostname=i1 "$IMAGE_PREFIX"cortex:$BACKWARD_IMAGE -target=ingester $INGESTER_ARGS -ingester.claim-on-rollout=true
+
+I1_ADDR=$(container_ip i1)
+wait_for "curl -s -f -m 3 $I1_ADDR/ready" "ingester ready"
+
+has_tokens_owned() {
+    curl -s -f -m 3 $1/metrics | grep -q 'cortex_ring_tokens_owned'
+}
+DIST_ADDR=$(container_ip distributor)
+wait_for "has_tokens_owned $DIST_ADDR" "distributor to see ingester in ring"
+
+docker run $RUN_ARGS --rm $AVALANCHE_IMAGE --metric-count=2 --label-count=2 --series-count=2 --remote-requests-count=1 --remote-url=http://distributor/api/prom/push
+
+echo Start second ingester, waiting for first
+docker run $RUN_ARGS -d --name=i2 --hostname=i2 "$IMAGE_PREFIX"cortex:$IMAGE_TAG -target=ingester $INGESTER_ARGS -ingester.join-after=300s -ingester.claim-on-rollout=true
+
+echo Stop first ingester so it should hand over
+docker stop i1
+
+I2_ADDR=$(container_ip i2)
+wait_for "curl -s -f -m 3 $I2_ADDR/ready" "ingester ready"
```

Comment:
How does this guarantee that the hand-over succeeded? Shouldn't we query back the second ingester and ensure it has the same series pushed to the first one?

Reply:
Some failures, e.g. continuous crashing, will be detected, but you are right that we could do a lot better in checking that the test was fully successful.

Reply:
I've extended the test to query back; it's still quite rudimentary.
I see in #1839 you did your own `Push()` rather than using Avalanche.  That would remove the problem that we can't check the random samples.

Reply:
Thanks for adding the querier. I understand the struggle of doing all this stuff with bash. The assertion you've introduced here LGTM (but please be aware tests are failing in CI).

This week I will revive the PR #1839 starting from your feedback (remove docker-compose, simplify it) and solve all TODOs: if we do a conjunct effort, me on the implementation and you on the review, we may get something solid soon, and start pushing people writing integration tests too when we do code reviews.